public_desktop/desktop-app.js

```diff
@@ -90,8 +92,60 @@ function startDesktopApp() {
 		}
 	}
 
+	function contextMenu( ev ) {
+		var menu = {};
+		var showEditorMenu = false;
+
+		// check if in visual editor, ipc sends an EventEmitter
+		if ( typeof ev.sender !== 'undefined' )  {
+			showEditorMenu = true;
+		}
+		// check if in textarea or similar
+		else if ( ( typeof ev.target !== 'undefined' ) && ev.target.closest( 'textarea, input, [contenteditable="true"]' ) ) {
+			showEditorMenu = true;
+		}
+
+		if ( showEditorMenu ) {
+			menu = buildEditorContextMenu(selection);
+		}
+		else {
+			var selectedText = window.getSelection().toString();
+			menu = buildGeneralContextMenu(selectedText);
+		}
+
+		// The 'contextmenu' event is emitted after 'selectionchange' has fired but possibly before the
+		// visible selection has changed. Try to wait to show the menu until after that, otherwise the
+		// visible selection will update after the menu dismisses and look weird.
+		setTimeout(function() {
+			menu.popup(electron.remote.getCurrentWindow());
+		}, 30);
+	}
+
+	function resetSelection() {
+		selection = {
+			isMisspelled: false,
+			spellingSuggestions: []
+		};
+	}
+
+
 	debug = gGebug( 'desktop:browser' );
 
+
+	debug( 'Setting up Context Menus' );
+	try {
+		var buildEditorContextMenu = electron.remote.require( '../desktop/lib/menu/editor-context-menu' );
+		var buildGeneralContextMenu = electron.remote.require( '../desktop/lib/menu/general-context-menu' );
```

Comment:
We could do something like this:

```
global.desktop = {
	config: config,
	settings: Settings,
	contextMenus: { // could also just be a wrapper lib that handles the requires
		editor: require( 'require( './lib/menu/editor-context-menu' ),
		general: require( 'require( './lib/menu/general-context-menu' ),
	}
};
```

and then with can just use `desktop.contextMenus.editor` instead of `buildEditorContextMenu`.